positional_encodings/positional_encodings.py

```diff
@@ -76,7 +76,40 @@ def forward(self, tensor):
         emb[:, :, : self.channels] = emb_x
         emb[:, :, self.channels : 2 * self.channels] = emb_y
 
-        return emb[None, :, :, :orig_ch].repeat(batch_size, 1, 1, 1)
+        return emb[None, :, :, :orig_ch]
+
+class FixedPositionalEncoding2D(nn.Module):
+    def __init__(self, shape, channels):
+        super(FixedPositionalEncoding2D, self).__init__()
+        self.pos_encoder = SinglePositionalEncoding2D(channels)
+        self.pos_encoding = self.pos_encoder(torch.ones(1, shape[0], shape[1], channels))
```

Comment:
This could probably work for all positional encodings! I could imagine the function call looking something like this:

```python3
# pos_encoder = PEnc2D(channels)
def __init__(self, pos_encoder, shape): #shape would include channels
  super(...)
  self.pos_encoder = pos_encoder
  self.pos_encoding = self.pos_encoder(torch.ones(1, **shape))
```

Reply:
Mmm this is indeed maybe even nicer. So have a wrapper class of sorts which you can use for all 3 methods.  Sounds good.

Reply:
I saw you added that, nice :smile_cat:

Reply:
Yep